stock_picking_order_alert/models/res_config_settings.py

```diff
@@ -0,0 +1,24 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    display_quantity_alert_percentage = fields.Boolean(
+        string="Display Quantity Alert Percentage",
+        default=False,
+        config_parameter="stock_picking_order_alert.display_quantity_alert_percentage",
+    )
+    quantity_alert_percentage = fields.Float(
+        string="Quantity Alert Percentage",
+        default=30.0,
+        config_parameter="stock_picking_order_alert.alert_percentage",
+        help="Percentage threshold for quantity alert in receipts. "
+        "Default is 30%. Set to 0 to disable alerts.",
+    )
+
+    @api.constrains("quantity_alert_percentage")
+    def _check_quantity_alert_percentage(self):
+        if self.quantity_alert_percentage <= 0:
+            raise UserError(_("Quantity alert percentage must be greater than zero."))
```

Comment:
```suggestion
    _sql_constraints = [("check_quantity_alert_percentage", "CHECK( display_quantity_alert_percentage AND quantity_alert_percentage>0 OR NOT display_quantity_alert_percentage)", "Quantity alert percentage must be greater than zero.")]
```

`_sql_constraints` should be more efficient than using a Python decorator.  
Additionally, validation should only be applied when `display_quantity_alert_percentage` is enabled, right?

Reply:
Im not sure if the values are stored at res.config level with `config_parameter` set, have you done this before @rrebollo?

Reply:
@Christian-RB my bad, you are absolutely right. My suggestion would maybe be valid in a company dependent design at the res.company level. But this feature is settle at server level ... I think ...